Data/ByteString.hs

```diff
@@ -1224,8 +1222,9 @@ intercalate (BS fSepPtr sepLen) (BS fhPtr hLen : t) =
             go (destPtr' `plusPtr` chunkLen) chunks
       go (dstPtr0 `plusPtr` hLen) t
   where
-  totalLen = List.foldl' (\acc (BS _ chunkLen) -> acc + chunkLen + sepLen) hLen t
-{-# INLINE [1] intercalate #-}
```

Comment:
Should not we retain `{-# INLINE intercalate #-}` (without phase number)?

Reply:
I think there's little to be gained from inlining this version of `intercalate`. Maybe when the list is known to be non-empty inlining can sometimes prevent allocation of a box for the separator? But even that will probably not happen very frequently; and if we want to facilitate that optimization there are better ways of doing so.

Reply:
I guess the `List.foldl'` will also fuse when `intercalate` is inlined and given an explicit list of length exactly two. But not much else can do so without an explicit `CONLIKE` pragma because `intercalate` needs to traverse the list twice.

Reply:
I'd put `{-# INLINABLE #-}` then and let GHC to decide.